diff --git a/tests/test_model_builders.py b/tests/test_model_builders.py
index 70a2e687..b82b053d 100644
--- a/tests/test_model_builders.py
+++ b/tests/test_model_builders.py
@@ -2184,7 +2184,7 @@ def test_logreg_builder_sequential_calls():
             ),
         ]
         if not sklearn_check_version("1.8")
-        else [(LogisticRegression(solver="liblinear"), 2)]
+        else []
     )
     + [
         # case below might change in the future if sklearn improves their modules
