diff --git a/sklearnex/tests/test_memory_usage.py b/sklearnex/tests/test_memory_usage.py
index 4e0ed87c..0b5074c2 100644
--- a/sklearnex/tests/test_memory_usage.py
+++ b/sklearnex/tests/test_memory_usage.py
@@ -280,6 +280,7 @@ def _kfold_function_template(
 @pytest.mark.parametrize("estimator", CPU_ESTIMATORS.keys())
 @pytest.mark.parametrize("data_shape", data_shapes)
 def test_memory_leaks(estimator, dataframe, queue, order, data_shape):
+    pytest.skip()
     func = ORDER_DICT[order]
     if estimator == "_assert_all_finite" and queue is not None:
         pytest.skip(f"{estimator} is not designed for device offloading")
